core/modules/user/js/user.js

```diff
@@ -166,9 +174,28 @@ Backdrop.evaluatePasswordStrength = function (password, username) {
   // its length. Again, adapted from zxcvbn.
   strength = (Math.log(cardinality) / Math.log(2)) * password.length + 1;
 
-  // Check if password is the same as the username.
-  if (password !== '' && password.toLowerCase() === username.toLowerCase()) {
-    strength = 5;
+  // Check if password is the same as the username or email.
+  if (password !== '') {
+    if (password === username || password === email) {
+      strength = 5;
+    }
+    // Consider admin password constraint settings if active.
+    if (config.constraints_active === "1") {
+      if (config.case_insensitive_comparison === 1) {
+        password = password.toLowerCase();
+        username = username.toLowerCase();
+        email = email.toLowerCase();
+      }
```

Comment:
@herbdool this way the case-insensitive comparison is applied only if constrains are active (`.toLowerCase()` conversion only happens when `config.case_insensitive_comparison === 1`). We should move the conditional with the `.toLowerCase()` conversion before the `if (password === username || password === email)`, so that it is applied regardless of the setting being "Enforce strong passwords" or "Indicate weak passwords":

```suggestion
    if (config.case_insensitive_comparison === 1) {
      password = password.toLowerCase();
      username = username.toLowerCase();
      email = email.toLowerCase();
    }
    if (password === username || password === email) {
      strength = 5;
    }
    // Consider admin password constraint settings if active.
    if (config.constraints_active === "1") {
```